src/org/daisy/dotify/api/writer/PagedMediaWriter.java

```diff
@@ -41,9 +43,10 @@ public interface PagedMediaWriter extends Closeable {
      * Inserts a new page in the output format,
      * if applicable.
      *
+     * @param page the page to write
      * @throws IllegalStateException if writer is not opened or if writer has been closed
      */
-    public void newPage();
+    public void newPage(Page page);
```

Comment:
I see two problems with this change:

- You use a class (`Page`) that is not part of the API (not in the `org.daisy.dotify.api.writer` package).
- Users of `PageMediaWriter` (I mean classes such as `WriterHandler`, not implementations of the interface) are now forced to pass `Page` objects. The problem with this is not that it is hard to implement the `Page` interface (it's just one method), but that it makes the API confusing: do rows need to be supplied through the `Page` objects or through `newRow` calls?

One possible solution could be to remove the `newRow` methods and let `newPage` add complete pages, however that would be inconsistent with the rest of the API (`newVolume` and `newSection` also don't add complete volumes/sections).

Is it not an option to use the `prepare` method?

Reply:
@bertfrees yes, I understand that the API change requires all implementations to change which I also did in the PR.

I might be missing something but the prepare function only have access to the metadata and that is not helpful if I want to extract the page number of the last page (the page with the highest number)

Am I missing something?

Best regards
Daniel

Reply:
When it comes to the Page interface we could extend it so it's less useless/confusing there are a lot of interesting information in the PageImpl class that we could expose to implementors of the Writers that could be useful.

Reply:
> I understand that the API change requires all implementations to change which I also did in the PR

Yes, all implementation of the `PageMediaWriter` interface and all classes such as `WriterHandler` that use `PageMediaWriter` to output a braille document. For both of these cases there could be code outside of the Dotify code that needs to be updated.

> I might be missing something but the prepare function only have access to the metadata

I was thinking the `prepare` function would be called by the formatter with the correct metadata. I think at the moment only the OBFL metadata is passed on, but it could be used for many other purposes. The writer implementation chooses which metadata is relevant and which isn't.

> When it comes to the Page interface we could extend it so it's less useless/confusing there are a lot of interesting information in the PageImpl class that we could expose to implementors of the Writers that could be useful.

Yes. Instead of `Page` we could perhaps pass an (optional) object that contains metadata about a page? A bit similar to `SectionProperties`, but different.